contract_payment_mode/models/contract.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, fields, models
+
+
+class ContractContract(models.Model):
+    _inherit = "contract.contract"
+
+    payment_mode_id = fields.Many2one(
+        comodel_name="account.payment.mode",
+        string="Payment Mode",
+        domain=[("payment_type", "=", "inbound")],
+        index=True,
+    )
+
+    @api.onchange("partner_id")
+    def on_change_partner_id(self):
```

Comment:
Changing this to compute method could be great (and change field to compute/store/readonly=False)